02-peer-protocol.md

```diff
@@ -445,6 +828,343 @@ to broadcast the commitment transaction to get his funds back and open a new
+out of an acceptable range, may fail the negotiation. However, it is
+recommended that the *accepter* permits the channel open to proceed
+without their participation in the channel's funding.
+
+Note that `open_channel`'s `channel_reserve_satoshi` has been omitted.
+Instead, the channel reserve is fixed at 1% of the total channel balance
+(`open_channel2`.`funding_satoshis` + `accept_channel2`.`funding_satoshis`)
+rounded down to the nearest whole satoshi or the `dust_limit_satoshis`,
+whichever is greater.
+
+Note that `push_msat` has been omitted.
+
+### The `accept_channel2` Message
+
+This message contains information about a node and indicates its
+acceptance of the new channel.
+
+1. type: 65 (`accept_channel2`)
+2. data:
+    * [`channel_id`:`channel_id`]
+    * [`u64`:`funding_satoshis`]
+    * [`u64`:`dust_limit_satoshis`]
+    * [`u64`:`max_htlc_value_in_flight_msat`]
+    * [`u64`:`htlc_minimum_msat`]
+    * [`u32`:`minimum_depth`]
+    * [`u16`:`to_self_delay`]
+    * [`u16`:`max_accepted_htlcs`]
+    * [`point`:`funding_pubkey`]
+    * [`point`:`revocation_basepoint`]
+    * [`point`:`payment_basepoint`]
+    * [`point`:`delayed_payment_basepoint`]
+    * [`point`:`htlc_basepoint`]
+    * [`point`:`first_per_commitment_point`]
+    * [`accept_tlvs`:`tlvs`]
+
+1. `tlv_stream`: `accept_tlvs`
+2. types:
+   1. type: 1 (`option_upfront_shutdown_script`)
+   2. data:
+       * [`u16`:`shutdown_len`]
+       * [`shutdown_len*byte`:`shutdown_scriptpubkey`]
+
+
+Rationale and Requirements are the same as listed above,
+for [`accept_channel`](#the-accept_channel-message) with the following
+additions.
+
+#### Requirements:
+
+The accepting node:
+    - MAY respond with a `funding_satoshis` value of zero.
+
+#### Rationale
+
+The `funding_satoshis` is the amount of bitcoin in satoshis
+the *accepter* will be contributing to the channel's funding transaction.
+
+Note that `accept_channel`'s `channel_reserve_satoshi` has been omitted.
+Instead, the channel reserve is fixed at 1% of the total channel balance
+(`open_channel2`.`funding_satoshis` + `accept_channel2`.`funding_satoshis`)
+rounded down to the nearest whole satoshi or the `dust_limit_satoshis`,
+whichever is greater.
+
+`channel_id` for the `accept_channel2` MUST be derived using a zero-d out
+basepoint for the peer's revocation basepoint.  This allows the peer to
+return channel-assignable errors before the *accepter*'s revocation
+basepoint is known.
+
+### Funding Composition
+Funding composition for channel establishment v2 makes use of the
+[Interactive Transaction Construction](#interactive-transaction-construction)
+protocol, with the following additional caveats.
+
+
+#### The `tx_add_input` Message
+
+No additional caveats or requirements.
+
+
+#### The `tx_add_output` Message
+##### Requirements
+The sending node:
+  - if is the *opener*:
+    - MUST send at least one `tx_add_output`,  which contains the
+      channel's funding output
```

Comment:
Maybe dumb question, but how does the `accepter` know which output is the funding output?
Don't we need to exchange its `serial_id` beforehand (for example it could be in `open_channel2`)?

Reply:
You can figure it out based on the funding_pubkeys that are exchanged in `open_channel2` and `accept_channel2`.

Reply:
You'd want to independently verify this anyway, instead of relying on your peer to tell you they did it.

Reply:
Good point, that was a dumb question! Sorry I'm new to this lightning thing :laughing: